### PR TITLE
Refactor js to ts/reza

### PR DESCRIPTION
**Pull request by kasurbantal**
1. Refactor components/addTodo/addTodo.jsx ke tsx
2. Refactor components/todocard/enum.js ke ts
3. Refactor components/todocard/todocard.jsx ke tsx
4. Menghapus component/Card
5. Memindahkan Component/Navbar ke folder components
6. Mengubah page.jsx menjadi page.tsx

### CODE REVIEW

File: todo-list-nextjs/src/components/addTodo/addTodo.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+}
+
+function AddTodoButton({ handleSubmit } : AddTodoButtonProps): React.ReactNode {
+  const [showInput, setShowInput] = useState<boolean>(false);
+  const [title, setTitle] = useState<string>("");
+  const [description, setDescription] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
 
-  const onTitleChangeHandler = (event) => {
+  const onTitleChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (indraseptama) [inline todo-list-nextjs/src/components/addTodo/addTodo.tsx:15]: Widih manteb 👍

File: todo-list-nextjs/src/components/addTodo/addTodo.tsx [hunk 2]
```diff
@@ -43,7 +47,7 @@ function AddTodoButton({ handleSubmit }) {
             onChange={onTitleChangeHandler}
             value={title}
             required
-          ></input>
+          />
```
Comment (indraseptama) [inline todo-list-nextjs/src/components/addTodo/addTodo.tsx:50]: nice

File: todo-list-nextjs/src/components/todocard/enum.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+enum TodoCardType {
```
Comment (indraseptama) [inline todo-list-nextjs/src/components/todocard/enum.ts:1]: export const saja, njagani nanti ada enum lainnya

File: todo-list-nextjs/src/components/todocard/todocard.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  activity: string;
+  deadline: string;
+  type: TodoCardType;
+  onComplete: (id: number) => void;
+  onInProgress: (id: number) => void;
+  onHold: (id: number) => void;
+  onDelete: (id: number) => void;
+}
+
+const TodoCard: React.FC<TodoCardProps> = ({ id, activity, deadline, type, onComplete, onInProgress, onHold, onDelete }) => {
```
Comment (indraseptama) [inline todo-list-nextjs/src/components/todocard/todocard.tsx:17]: mantabb sebenarnya bisa gini za, tapi bebas

`const TodoCard = ({ id, activity, deadline, type, onComplete, onInProgress, onHold, onDelete } : TodoCardProps)`

**Review by indraseptama — APPROVED** [review]
lgtm with minor comment

File: todo-list-nextjs/src/components/todocard/enum.ts (hunk 3 above)
Comment (kasurbantal) [reply todo-list-nextjs/src/components/todocard/enum.ts:1]: kalo dibuat  const keluar error terus coach

File: todo-list-nextjs/src/components/addTodo/addTodo.tsx (hunk 2 above)
Comment (kasurbantal) [reply todo-list-nextjs/src/components/addTodo/addTodo.tsx:50]: thx coach

File: todo-list-nextjs/src/components/addTodo/addTodo.tsx (hunk 1 above)
Comment (kasurbantal) [reply todo-list-nextjs/src/components/addTodo/addTodo.tsx:15]: thx coach

File: todo-list-nextjs/src/app/todo-app/page.jsx [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import { usePage } from "./usePage";
 import styles from "./styles.module.css";
 import TodoCard from "../../components/todocard/todocard";
-import { TodoCardType } from "../../components/todocard/enum";
+import TodoCardType from "../../components/todocard/enum";
```
Comment (indraseptama) [inline todo-list-nextjs/src/app/todo-app/page.jsx:6]: kalo diganti export const, ini gaperlu ada perubahan. jadi tetap impor {TodoCardType}

Comment (kasurbantal) [reply todo-list-nextjs/src/app/todo-app/page.jsx:6]: ok coach, saya obok2 lagi later

**Pull request merged**